File: 프로그래머스/unrated/181894. 2의 영역/2의 영역.py
def solution(arr):
    data = []
    for i in range(len(arr)):
        if arr[i] == 2:
            data.append(i)
    
    if not data:
        return [-1]
    if len(data) == 1:
        return [2]
    return arr[data[0]:data[-1]+1]

File: 프로그래머스/0/181894. 2의 영역/2의 영역.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        int s = -1;
        int e = -1;
        
        for (int i=0; i<arr.length; i++) {
            if(arr[i] == 2) {
                if(s == -1) {
                    s = i;
                    e = i;
                } else {
                    e = i;
                }
            }
        }
        
        if(s == -1) {
            return new int[] { -1 };
        }
        return Arrays.copyOfRange(arr, s, e+1);
    }
}